### DISCUSSION TITLE
Smartmeter Type C how to decode

### DISCUSSION THREAD

**Discussion by Marty56** [arendst/Tasmota#15157] [Q&A] [answered]
Hi, 

I have a meter that is creating 400 pulses every 1kWh. So I thought I would use a type c smart meter.
Unfortunately I do not find or understand the syntax of the decode section.

```>D
>B
=>sensor53 r
>M 1
+1,16,c,0,-50,Strom
1,1-0:1.8.0*255(@400+7631.400,Strom,kWh,Auto,2
#
```

To come up with the non working approach, I copied the examples given without understanding how the decode section is working.

I have a Wemos D1 mini and have my S0 Input attach to pin D0 which should be GPIO 16.
It has no pullup and should count with a debouncing of 50 sec. This is the "+1,16,c,0,-50,Strom" line. So far so good.
But then in the next line after the @ sign it should increase the kWh very 400 pulses and I am using an offset of 7631 pulse to reflect the actual value of the meter display. But what does "1-0:1.8.0*255(" mean which I have seen in examples?

Can anyone provide me with a tip?

regards Martin

**Comment by gemu2015** [comment, marked as answer]
in counter mode the Tasmota persistent counter is used as counter reference. so better set the counter with sensor53 c1 7631 (see docs) and do not use the offset in the decode line. (this is meant for uncalibrated power meters)
the "1-0:1.8.0*255(" is a constant string used for decoding of counter values actually that it is also an OBIS code.
depending on your input you may also need a pull-up resistor or enable the build in pull-up (see docs)

gpio16 does not support IRQ counter input, so better use another pin e.g. gpio2

so this should probably work with your input 

in console: sensor53 c1 7631
however if you want to set kWh you may have to multiply by 400 because the decode line devides through 400

\>B
=>sensor53 r
\>M 1
+1,2,c,0,-50,Strom
1,1-0:1.8.0*255(@400,Strom,kWh,Auto,2
#

there is an exhaustive thread about the SML driver in this german forum https://forum.creationx.de/forum/index.php?thread/1095-d0-zähler-sml-auslesen-mit-tasmota/

**Comment by Marty56** [comment]
Thanks this information was helpful. It seems to work now! Great.
